go/ct/evm_fuzz_test.go

```diff
@@ -0,0 +1,274 @@
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package ct_test
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"pgregory.net/rand"
+
+	"github.com/Fantom-foundation/Tosca/go/ct"
+	. "github.com/Fantom-foundation/Tosca/go/ct/common"
+	"github.com/Fantom-foundation/Tosca/go/ct/st"
+	"github.com/Fantom-foundation/Tosca/go/vm"
+	"github.com/Fantom-foundation/Tosca/go/vm/geth"
+	"github.com/Fantom-foundation/Tosca/go/vm/lfvm"
+)
+
+// FuzzGeth is a fuzzing test for the geth EVM implementation
+// TODO: it would be interesting to have a method to extend the corpus of the fuzzer
+// with the failures found in other tests.
+// So far failures will be stored in the folder: testdata/fuzz/FuzzerFunctionName/
+func FuzzGeth(f *testing.F) {
+	fuzzVm(geth.NewConformanceTestingTarget(), f)
+}
+
+// FuzzLfvm is a fuzzing test for lfvm
+func FuzzLfvm(f *testing.F) {
+	fuzzVm(lfvm.NewConformanceTestingTarget(), f)
+}
+
+// FuzzLfvm is a fuzzing test for evmzero
+// func FuzzEvmzero(f *testing.F) {
+// 	fuzzVm(evmzero.NewConformanceTestingTarget(), f)
+// }
+
+// FuzzDifferentialLfvmVsGeth compares state output between lfvm and geth
+func FuzzDifferentialLfvmVsGeth(f *testing.F) {
+	differentialFuzz(f,
+		lfvm.NewConformanceTestingTarget(),
+		geth.NewConformanceTestingTarget(),
+	)
+}
+
+// TODO: #549 this test makes sense but cannot be enabled yet:
+// - The evmzero fails the differential test with geth. #54
+// - Any other invocation of fuzzing tests in this file seem to
+// invoke the all other tests in the file, and they will fail.
+// func FuzzDifferentialEvmzeroVsGeth(f *testing.F) {
+// 	differentialFuzz(f,
+// 		evmzero.NewConformanceTestingTarget(),
+// 		geth.NewConformanceTestingTarget(),
+// 	)
+// }
+
+//////////////////////////////////////////////////////////////////////////////
+// Fuzzing helpers
+
+func differentialFuzz(f *testing.F, testeeVm ct.Evm, referenceVm ct.Evm) {
+
+	rnd := rand.New(0)
+
+	prepareFuzzingSeeds(f, rnd)
+
+	f.Fuzz(func(t *testing.T, opCodes []byte, gas int64, revision byte, stackBytes []byte) {
+
+		state, err := corpusInputDataToCtState(opCodes, gas, revision, stackBytes)
+		if err != nil {
+			t.Skip(err)
+		}
+
+		testeeResultState, err := testeeVm.StepN(state.Clone(), 1)
+		defer testeeResultState.Release()
+		if err != nil {
+			t.Fatalf("failed to run test case: %v", err)
+		}
+
+		referenceResultState, err := referenceVm.StepN(state.Clone(), 1)
+		defer referenceResultState.Release()
+		if err != nil {
+			t.Fatalf("failed to run test case in reference VM: %v", err)
+		}
+
+		if testeeResultState.Status != referenceResultState.Status {
+			t.Fatal("invalid result, status does not match reference status:", errorReportString(state, testeeResultState, referenceResultState))
+		}
+
+		// if result is other than running, further checks may be misleading
+		if testeeResultState.Status != st.Running {
```

Comment:
I would not consider `Failure` same as `Returned` or `Revert`. Perhaps this check can be a little more specialized

Reply:
Good idea, unfortunately the Fuzzer fails at the moment, and I can not do this change and test it. I comment on it, and we can expand the fuzzer later on.